.github/actions/build-infinispan/action.yml

```diff
@@ -0,0 +1,37 @@
+name: Build Infinispan
+description: Builds Infinispan and Maven repository with all required artifacts
+
+runs:
+  using: composite
+  steps:
+    - name: Setup Java
+      uses: ./.github/actions/setup-java
+
+    - name: Build Infinispan
+      shell: bash
+      run: |
+        # Ensure this plugin is built first to avoid warnings in the build
+        ./mvnw install -Pdistribution -s maven-settings.xml
+        # By using "dependency:resolve", it will download all dependencies used in later stages for running the tests
+        ./mvnw install dependency:resolve -V -e -DskipTests -DexcludeGroupIds=org.infinispan -Dsilent=true -s maven-settings.xml
```

Comment:
I don't understand this step. We are archiving only `.m2/repository/org/infinispan` but `dependency:resolve` will download other dependencies.

Reply:
The `setup-java` also caches maven dependencies, so  what happens in the subsequent jobs is that the `setup-java` cache is restored to `~/.m2` and then we explicitly overwrite the `org/infinispan` artifacts with `./.github/actions/extract-maven-artifacts`. 

This is a similar approach to Keycloak, however in their CI they don't use the built in cache of `setup-java`. Instead they manually create the m2 cache using the `actions/cache` and explicitly exclude `org/keycloak` dependencies from this. This is more verbose, but the advantage of this approach is that it means that a PR which still has a reference to a removed module will never pass.

The  more I think about it, the more I think we should be safe and follow the same approach as Keycloak.